### DISCUSSION TITLE
Rename

### DISCUSSION THREAD

**Discussion by ghost** [sacalon/sacalon#57] [Ideas]
**Is your feature request related to a problem? Please describe.**
The name "Hascal" is extremely similar to the programming language "Haskell" and "Pascal" somewhat. This was probably intended, but it is extremely confusing and (I think) it may prevent wide adoption.

**Describe the solution you'd like**
Rename the language to something more unique
**Describe alternatives you've considered**

**Additional context**

**Comment by bistcuite** [comment]
What name do you think is appropriate?

**Comment by ghost** [comment]
I'm not very creative, but I think that if the name was more unique then it could be seen more. Nim and Zig are two examples, they have extremely unique names that haven't been used in other things.

**Comment by mehanalavimajd** [comment]
> **Is your feature request related to a problem? Please describe.** The name "Hascal" is extremely similar to the programming language "Haskell" and "Pascal" somewhat. This was probably intended, but it is extremely confusing and (I think) it may prevent wide adoption.
> 
> **Describe the solution you'd like** Rename the language to something more unique **Describe alternatives you've considered** A clear and concise description of any alternative solutions or features you've considered.
> 
> **Additional context** Add any other context or screenshots about the feature request here.

Extremely agreed 👍

**Comment by mehanalavimajd** [comment]
> What name do you think is appropriate?

Isn't sun a good name?
- Sun programming language

**Comment by bistcuite** [comment]
> > What name do you think is appropriate?
> 
> Isn't sun a good name?
> 
> * Sun programming language

*Sun* is a good suggestion but we need more names, to choose best name.

**Comment by bistcuite** [comment]
Some good names :
- H
- Holt
- Tesla
- Galaxy
- Nova
- Xarazmi
- Halo

 I think *Halo* in this list is more beautiful than the other members of the list.

**Comment by mehanalavimajd** [comment]
> Some good names :
> 
> * H
> * Holt
> * Tesla
> * Galaxy
> * Nova
> * Xarazmi
> * Halo
> 
> I think _Halo_ in this list is more beautiful than the other members of the list.

I love nova

**Comment by bistcuite** [comment]
> I love nova

Unfortunately nova [exists](http://www.navgen.com/nova/).

**Comment by mehanalavimajd** [comment]
Halo is also name of a tv series.
H is good

**Comment by bistcuite** [comment]
> Halo is also name of a tv series. H is good

I [found](https://h.christine.website/) a language with name "H", but there is no problem in choosing H, because projects of the same name with H are half-finished, they are also unknown.
Also @h-lang and @hlang user names is not available in GitHub.

**Comment by ghost** [comment]
> Some good names :
> - H
> - Holt
> - Tesla
> - Galaxy
> - ~~[Nova](http://www.navgen.com/nova/)~~
> - Xarazmi
> - Halo
> 
>  I think *Halo* in this list is more beautiful than the other members of the list.

I think nova could still be an option since the language you referenced seemed pretty old and obscure. I also think that it might be better if the name reflects the language's goals and what it can do, kind of like what hascal is right now(the language being influenced by Pascal/Haskell)

**Comment by bistcuite** [comment]
> 
That's right👍

**Comment by mmdbalkhi** [comment]
I think it is better if it is a noble name with Persian roots, or if this name is borrowed from the legends of Persian...

**Reply by bistcuite** [reply]
Do you have any idea about it? What's your opnion about "Rostam"?

**Reply by mmdbalkhi** [reply]
Some suggested names in my opinion:

- Ahriman
- Ahura
- Hooshang
- kiyomarth
- Kian
- Siamak
- [etc](https://fa.wikipedia.org/wiki/%D8%A7%D8%B3%D8%B7%D9%88%D8%B1%D9%87%E2%80%8C%D8%B4%D9%86%D8%A7%D8%B3%DB%8C_%D8%A7%DB%8C%D8%B1%D8%A7%D9%86%DB%8C)
